lib/content-services/src/lib/viewer/components/alfresco-viewer.component.ts

```diff
@@ -348,13 +349,11 @@ export class AlfrescoViewerComponent implements OnChanges, OnInit {
 
                 const nodeMimeType = nodeData?.content?.mimeType;
                 const renditionMimeType = nodeRendition.mimeType;
-                mimeType = renditionMimeType || nodeMimeType;
+                this.mimeType = renditionMimeType || nodeMimeType;
```

Comment:
Can we cover that with some small unit test case check?

Reply:
I'm not sure if a unit test would make a difference here really. If you see in the earlier code, `this.mimetype` would have anyways gotten updated in the same function execution in line num 357. We only added an assignment earlier in the function execution to make sure it gets updated in the same change detection cycle as the file name.

Reply:
I think with https://github.com/Alfresco/alfresco-ng2-components/pull/10508 this PR might be obsolete, currently when renditions are generated with this approach the icon is changed twice and ultimately is not correct, the other fixes both issues